### PR TITLE
Fix missing docs in time series source reader

### PR DESCRIPTION
**Pull request by dnhatn**
While working on rate aggregation, I noticed that the time series source might miss documents. Specifically, when we reach the [max_page_size](https://github.com/elastic/elasticsearch/pull/106705/files#diff-8bb21ea96e1c7ab1c4e97c1ad295414b084f9703c335962e740ddff28ff00c72L271)  limit and the tsid [changes](https://github.com/elastic/elasticsearch/pull/106705/files#diff-8bb21ea96e1c7ab1c4e97c1ad295414b084f9703c335962e740ddff28ff00c72L274), we skip that document because we call  [nextDoc](https://github.com/elastic/elasticsearch/pull/106705/files#diff-8bb21ea96e1c7ab1c4e97c1ad295414b084f9703c335962e740ddff28ff00c72L266) again when resuming reading. Also, I think this operator should honor the max_page_size limit and avoid emitting pages that exceed this threshold.

Labelled this non-issue for an unreleased bug.

### CODE REVIEW

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/TimeSeriesSortedSourceOperatorFactory.java [hunk 1]
```diff
@@ -228,62 +228,43 @@ protected boolean lessThan(Leaf a, Leaf b) {
...
                         Leaf leaf = queue.top();
                         segmentsBuilder.appendInt(leaf.segmentOrd);
                         docsBuilder.appendInt(leaf.iterator.docID());
                         timestampIntervalBuilder.appendLong(leaf.timestamp);
                         tsOrdBuilder.appendInt(globalTsidOrd);
+                        final Leaf newTop;
                         if (leaf.nextDoc()) {
                             // TODO: updating the top is one of the most expensive parts of this operation.
                             // Ideally we would do this a less as possible. Maybe the top can be updated every N docs?
-                            Leaf newTop = queue.updateTop();
```
Comment (martijnvg) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/TimeSeriesSortedSourceOperatorFactory.java:250]: The idea behind this logic is that all docs of a time serie are always contained within a page. Downstream operators could optimize for that, currently this doesn't happen and the grouping operator in the other pr doesn't either (yet). So I'm okay with simplifying this for now and at the same time fixing the bug :)

**Review by martijnvg — APPROVED** [review]
👍

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/TimeSeriesSortedSourceOperatorFactory.java (hunk 1 above)
Comment (dnhatn) [reply x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/TimeSeriesSortedSourceOperatorFactory.java:250]: Yes, I understand that idea and we will re-enable this short-cut by using ConstantBlocks.

**Comment by dnhatn** [comment]
Thanks @martijnvg!

**Pull request merged**